File: Java Implementation/src/P21.java
public class P21 {

  public static void main(String[] args) {

    int sumOfAmicableNumbers = 0;

    for (int i = 1; i <= 10000; i++) {
      int j = sumOfDivisors(i);
      if ((sumOfDivisors(j) == i) && (i != j)) {
        sumOfAmicableNumbers += i;
        System.out.println("Amicable pair found: " + i + ", " + j);
      }
    }

    System.out.println("Sum: " + sumOfAmicableNumbers);
  }

  public static int sumOfDivisors(int n) {
    int result = 0;
    for (int i = 1; i <= (n/2); i++) {
      if (n % i == 0) {
        result += i;
      }
    }
    return result;
  }

}

File: Swift Implementation/P21.swift
//
//  P21.swift
//  
//
//  Created by Xin Jet Chew on 07/01/2017.
//
//

import Foundation
import Darwin

func sumOfDivisors(n: Int) -> Int {
    var sum: Int = 0
    if (n > 1) {
        for i in 1 ... n/2 {
            if n % i == 0 {
                sum += i
            }
        }
    }
    return sum
}

let limit = 10000

var sumOfAmiciablePairs: Int = 0

for a in 1 ... limit {
    let b = sumOfDivisors(n: a)
    if sumOfDivisors(n: b) == a && a != b {
        print("Amicable pair found: \(a), \(b)")
        sumOfAmiciablePairs += a
    }
}

print(sumOfAmiciablePairs)
